### PR TITLE
Feature/add rsvp status

### PR DESCRIPTION
**Pull request by RobertWSaunders**
adding rsvp logic among other things to dashboard

### CODE REVIEW

File: client/components/Dashboard/Dashboard.js [hunk 1]
```diff
@@ -109,21 +92,29 @@ class Dashboard extends Component {
...
+            return (
+                <Confetti {...this.props.size} numberOfPieces={400} recycle={false} />
+            );
+        }
+        return null;
+    }
+
     render() {
+        const { width, height } = this.props.size;
+        console.log(width, height);
```
Comment (zackharley) [inline client/components/Dashboard/Dashboard.js]: Remove console.log

File: client/components/Forms/RSVPForm.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    }
+
+    renderRSVPFormErrorMessage() {
+        return (
+            <Message
+                error
+                size='small'
+                className="error-message"
+                header='Something went wrong!'
+                content='Oops! Something has gone wrong, please try again.'
```
Comment (zackharley) [inline client/components/Forms/RSVPForm.js]: Can we add an error message here?

File: client/components/Hacker/ApplicationWithdrawn.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { Header, Segment } from 'semantic-ui-react';
+import Emoji from 'react-emoji-render';
+import React from 'react';
+
+export default () => (
+    <Segment>
+        <Header as='h2'><Emoji text="Thank you from QHacks! :heart:"/></Header>
+        <p>While its too bad that you wont be joining us, we want to say thank you for applying! We understand that things come up last minute, just know that we loved your application and that someone on the waiting list will be happy that you have chosen to withdraw your position.</p>
```
Comment (zackharley) [inline client/components/Hacker/ApplicationWithdrawn.js]: `wont` --> `won't`

Comment (zackharley) [inline client/components/Hacker/ApplicationWithdrawn.js]: `we want to say thank you for applying.` --> `thank you for withdrawing your application.`

Comment (zackharley) [inline client/components/Hacker/ApplicationWithdrawn.js]: `We understand that things come up last minute, just know that we loved your application and that someone on the waiting list will be happy that you have chosen to withdraw your position.` --> `Someone from the waitlist will now get to attend QHacks 2018.` Maybe? This is kind of a strange thing to tell someone.

File: client/components/Hacker/DeclinedApplicant.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { Header, Segment } from 'semantic-ui-react';
+import Emoji from 'react-emoji-render';
+import React from 'react';
+
+export default () => (
+    <Segment>
+        <Header as='h2'><Emoji text="Thank you from QHacks! :heart:"/></Header>
+        <p>This years applicant pool was the best yet, making our decision-making process harder than ever. Unfortunately, we will be unable to offer you a spot at QHacks 2018.</p>
+        <p>That being said, dont let this get you down! There are plenty of other great hackathons that you can attend. Check out MLH for more information. Make sure to apply again next year for QHacks 2019.</p>
```
Comment (zackharley) [inline client/components/Hacker/DeclinedApplicant.js]: Wrap MLH in `<a>`

Comment (zackharley) [inline client/components/Hacker/DeclinedApplicant.js]: `dont` --> `don't`

File: client/components/Hacker/DeclinedApplicant.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+        <p>We are also looking for volunteers to help with event logistics for QHacks. This is a great opportunity to meet other people interested in technology and innovation, as well as getting most of the same benefits as hackers at the event. To be a volunteer, please fill our this Typeform.</p>
```
Comment (zackharley) [inline client/components/Hacker/DeclinedApplicant.js]: Wrap `this TypeForm` in `<a>`

File: client/components/Hacker/DeclinedApplicant.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+        <p>If you have any questions, you can reach out to us at hello@qhacks.io. Keep hacking!</p>
```
Comment (zackharley) [inline client/components/Hacker/DeclinedApplicant.js]: Wrap email in `<a>`

File: client/components/Hacker/WaitlistedApplicant.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { Header, Segment } from 'semantic-ui-react';
+import Emoji from 'react-emoji-render';
+import React from 'react';
+
+export default () => (
+    <Segment raised>
+        <Header as='h2'><Emoji text="Thank you from QHacks! You are on the waitlist! :heart:"/></Header>
+        <p>This years applicant pool was the best yet, making our decision-making process harder than ever. Unfortunately, we will be unable to offer you a spot at QHacks 2018.</p>
```
Comment (zackharley) [inline client/components/Hacker/WaitlistedApplicant.js]: This is the declined hacker text

File: client/components/Login/Login.js [hunk 8]
```diff
@@ -61,7 +61,7 @@ class Login extends Component {
                                 className="fontWeight-lighter"
                                 inverted
                                 size="medium">
-                            QHacks 2018, Queen's University
+                            QHacks 2018, Queens University
```
Comment (zackharley) [inline client/components/Login/Login.js]: Should be `Queen's`

File: client/components/MenuBar/MenuBar.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
+import { Menu, Button, Dropdown } from 'semantic-ui-react';
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+
+class MenuBar extends Component {
+
+    renderQHacksCrown() {
+        return (
+            <svg width="77px" height="40px" viewBox="0 0 77 40" version="1.1" xmlns="http://www.w3.org/2000/svg">
```
Comment (zackharley) [inline client/components/MenuBar/MenuBar.js]: Why is the SVG code being returned?

File: client/components/utils/AuthSwitch.js [hunk 10]
```diff
@@ -2,23 +2,19 @@ import { Switch } from 'react-router-dom';
...
+        if (!authType) throw Error('You cannot use <AuthSwitch> without specifying an auth type!');
 
         const childs = [];
         React.Children.forEach(children, (element) => {
             if (React.isValidElement(element)) {
-                const { type: elementAuthType } = element.props;
+                const { types: elementAuthTypes } = element.props;
 
-                if (element.type !== PrivateRoute || (element.type === PrivateRoute && elementAuthType === authType)) {
+                if (element.type !== PrivateRoute || (element.type === PrivateRoute && elementAuthTypes.includes(authType))) {
```
Comment (zackharley) [inline client/components/utils/AuthSwitch.js]: Great addition

**Comment by zackharley** [comment]
Need to correct my changes when we have a second

File: client/components/Forms/RSVPForm.js (hunk 2 above)
Comment (RobertWSaunders) [reply client/components/Forms/RSVPForm.js]: Yep, was going to address error handling after solid base is created.

File: client/components/Hacker/ApplicationWithdrawn.js (hunk 3 above)
Comment (RobertWSaunders) [reply client/components/Hacker/ApplicationWithdrawn.js]: Stupid atom glitch, I gotta fix that.

Comment (RobertWSaunders) [reply client/components/Hacker/ApplicationWithdrawn.js]: Text is just filler.

Comment (RobertWSaunders) [reply client/components/Hacker/ApplicationWithdrawn.js]: Text is just filler, didn't really put too much effort here.

File: client/components/Hacker/DeclinedApplicant.js (hunk 4 above)
Comment (RobertWSaunders) [reply client/components/Hacker/DeclinedApplicant.js]: Yep, was going to revisit text later.

File: client/components/Hacker/DeclinedApplicant.js (hunk 5 above)
Comment (RobertWSaunders) [reply client/components/Hacker/DeclinedApplicant.js]: Yep, was going to revisit text later.

File: client/components/Hacker/DeclinedApplicant.js (hunk 6 above)
Comment (RobertWSaunders) [reply client/components/Hacker/DeclinedApplicant.js]: Yep, was going to revisit text later.

File: client/components/Hacker/WaitlistedApplicant.js (hunk 7 above)
Comment (RobertWSaunders) [reply client/components/Hacker/WaitlistedApplicant.js]: Yep, was going to revisit text later.

File: client/components/MenuBar/MenuBar.js (hunk 9 above)
Comment (RobertWSaunders) [reply client/components/MenuBar/MenuBar.js]: Inline allows for height and width to be represented properly in menu, if you do a require it restricts size.

File: client/components/Hacker/ArriveInformation.js [hunk 11]
```diff
@@ -0,0 +1,60 @@
+import { Header, Segment, Card, Item, Image } from 'semantic-ui-react';
+import Emoji from 'react-emoji-render';
+import React from 'react';
+
+export default () => (
+    <div>
+        <Segment>
+            <Header as='h2'><Emoji text="Your spot is confirmed! :smiley: :fire:"/></Header>
+            <p>Yay! Its confirmed, you're coming to QHacks 2018. Now, what can you do in the meantime? Start thinking about stuff you'd like to build and make sure you do all of the stuff listed below, that way you can stay the most informed.</p>
+            <p>As outlined in your acceptance email we have busses traveling from Toronto, Waterloo, Montreal and Ottawa. In order to get on the bus that works for you, it is your responsibility to email <a href="mailto:travel@qhacks.io">travel@qhacks.io</a> to secure your spot. The busses will be filled on a first-come-first-serve basis. If you cannot get on a bus because one is not leaving from your location we will be offering reimbursements on a case-by-case basis, to figure this out please also email <a href="mailto:travel@qhacks.io">travel@qhacks.io</a>.</p>
```
Comment (zackharley) [inline client/components/Hacker/ArriveInformation.js]: Change to:
```
<p>As outlined in your acceptance email we have busses traveling from Toronto, Waterloo, Montreal and Ottawa. In order to reserve a spot on one of the buses, you must email <a href="mailto:travel@qhacks.io">travel@qhacks.io</a>. The busses will be filled on a first-come-first-serve basis, so make sure to send us an email as soon as possible.</p>

<p>If you are travelling to QHacks from another location than the ones listed above, we will be offering travel reimbursements. Please email <a href="mailto:travel@qhacks.io">travel@qhacks.io</a> for more information.</p>
```

File: client/components/Hacker/ArriveInformation.js [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+            <p>A few things to note before you arrive:</p>
+            <ul>
+                <li>Teams are allowed to be groups of 1 to 4 people.</li>
+                <li>You have 36 hours to build whatever you desire.</li>
+                <li>Registration and dinner will run from 5:00 PM to 8:00 PM on Friday, February 2nd. A full schedule will be released in the coming days along with a hacker package outlining some important information.</li>
+                <li>Due to space constraints we will have overflow rooms for our opening ceromonies that will stream what is happening live, it is in your best interest to arrive early. We are very forutnate this year to have some amazing speakers for our fireside chat, check them out on our <a href="http://qhacks.io/">website</a>.</li>
+                <li>You're going to have a blast!</li>
+            </ul>
+            <p>Finally, we want to say thank you for your interest, we look forward to hosting you for an exciting weekend. If you have any questions at all, pleast don't hesitate to email <a href="mailto:hello@qhacks.io">hello@qhacks.io</a>.</p>
```
Comment (zackharley) [inline client/components/Hacker/ArriveInformation.js]: <p>We can't wait to see you at QHacks! If you have any questions at all, please don't hesitate to email <a href="mailto:hello@qhacks.io">hello@qhacks.io</a>.</p>

File: client/components/Hacker/ArriveInformation.js [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+                <Item>
+                   <Item.Image size='mini' src={require('../../assets/img/slack.svg')} />
+                   <Item.Content verticalAlign='middle'>
+                     <Item.Header as='a' href="https://join.slack.com/t/qhacks2018/shared_invite/enQtMjk5NDMxODk0NzIxLTY5NjE1NjliMGMzODU5MDM0ZmY2MmRlMDI1YTE1ZWFhOTczMDUwODFiZjVjZDE1ODA4ZGY0NjAyYWJhNzYyMzM">Join the QHacks 2018 Slack!</Item.Header>
+                     <Item.Description>
+                        This will be the primary method our team as well as our partners will communicate with everyone before and during the event! Make sure to join to stay up to date.
+                     </Item.Description>
+                   </Item.Content>
+                 </Item>
+                 <Item>
```
Comment (zackharley) [inline client/components/Hacker/ArriveInformation.js]: Remove Facebook group

File: client/components/Hacker/ArriveInformation.js [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+                        This is an alternative to Slack, we will try our best to make the same posts in both places but cannot guarantee anything, its best to be on both.
+                      </Item.Description>
+                    </Item.Content>
+                  </Item>
+                  <Item>
+                     <Item.Image size='mini' src={require('../../assets/img/devpostLogo.png')} />
+                     <Item.Content verticalAlign='middle'>
+                       <Item.Header as='a' href="https://qhacks2018.devpost.com/">Check out the QHacks 2018 Devpost!</Item.Header>
+                       <Item.Description>
+                          This is where you will post the projects you will be working on so that we can judge them, so get aquainted with it. More information like prizes, judges and our partners will be posted here in the coming days.
```
Comment (zackharley) [inline client/components/Hacker/ArriveInformation.js]: `Devpost is where you will submit your projects at the end of the weekend. More information like prizes, judges and our partners will be posted here in the coming days.`

File: client/components/Hacker/ArriveInformation.js [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+                       </Item.Description>
+                     </Item.Content>
+                  </Item>
+                  <Item>
+                     <Item.Image size='mini' src={require('../../assets/img/twitter.svg')} />
+                     <Item.Content verticalAlign='middle'>
+                       <Item.Header as='a' href="https://twitter.com/qhacks18?lang=en">Follow QHacks on Twitter for giveaways!</Item.Header>
+                       <Item.Description>
+                            We will be having numerous giveaways on Twitter before and during the event, so make sure you give us a follow, trust us when we say you wouldn't want to miss out.
```
Comment (zackharley) [inline client/components/Hacker/ArriveInformation.js]: `We will be having numerous giveaways on Twitter before and during the event, so make sure you give us a follow, trust us when we say you wouldn't want to miss out.` ---> `... give us a follow. You won't want to miss out!`

File: client/components/Hacker/SuccessfulApplicant.js [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+            return (
+                <ArriveInformation />
+            );
+        }
+        return (
+            <div>
+
+                <Message positive>
+                    <Message.Header><Emoji text="Congratulations, you've been accepted to QHacks! :tada:"/></Message.Header>
+                    <p>Please RSVP as soon as possible to secure your spot. If you no longer wish or can come, please withdraw your application to open your spot to someone else on the waiting list.</p>
```
Comment (zackharley) [inline client/components/Hacker/SuccessfulApplicant.js]: `If you no longer wish or can come,` --> `If you don't plan on coming,`

**Pull request merged**